### ISSUE TITLE
Error downloading from BBC website: 'no suitable InfoExtractor'

### ISSUE THREAD

**Issue by azev** [ytdl-org/youtube-dl#23660]
Error downloading from BBC website:
```
youtube-dl "https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness"
[bbc] how-positive-thinking-is-harming-your-happiness: Downloading webpage
ERROR: no suitable InfoExtractor for URL https://www.bbc.co.uk/programmes/None
```

**Comment by Danieldavid521** [comment]
There a tons of similar issues on this repo for the same error, I just don't think youtube-dl is ment for generic website video downloading.

**Comment by Vangelis66** [comment]
... As said, this is just a duplicate of, e.g., https://github.com/ytdl-org/youtube-dl/issues/21870 😉 
Easy workaround is to first extract the clip's `PID` from the original URL, compose the corresponding iPlayer `programmes` URI and feed the latter to yt-dl: 

Reel URL: https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness
Clip PID: **p07c6sb6**
`Programmes` URI: https://www.bbc.co.uk/programmes/p07c6sb6

`youtube-dl -F https://www.bbc.co.uk/programmes/p07c6sb6` =>
```
[bbc.co.uk] p07c6sb6: Downloading video page
[bbc.co.uk] p07c6sb9: Downloading media selection XML
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
[bbc.co.uk] p07c6sb9: Downloading MPD manifest
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
[bbc.co.uk] p07c6sb9: Downloading MPD manifest
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
[bbc.co.uk] p07c6sb9: Downloading MPD manifest
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
[bbc.co.uk] p07c6sb9: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
[bbc.co.uk] p07c6sb9: Downloading MPD manifest
[info] Available formats for p07c6sb9:
format code
                   extension  resolution note
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_dash-audio_eng_
1=128000           m4a        audio only [en] DASH audio  128k , m4a_dash contai
ner, mp4a.40.2 (48000Hz)
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_dash_https-audi
o_eng_1=128000     m4a        audio only [en] DASH audio  128k , m4a_dash contai
ner, mp4a.40.2 (48000Hz)
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_dash-audio_e
ng_1=128000        m4a        audio only [en] DASH audio  128k , m4a_dash contai
ner, mp4a.40.2 (48000Hz)
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_dash_https-a
udio_eng_1=128000  m4a        audio only [en] DASH audio  128k , m4a_dash contai
ner, mp4a.40.2 (48000Hz)
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_dash-video=8270
00                 mp4        704x396    DASH video  827k , mp4_dash container,
avc3.4D401E, 25fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_dash_https-vide
o=827000           mp4        704x396    DASH video  827k , mp4_dash container,
avc3.4D401E, 25fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_dash-video=8
27000              mp4        704x396    DASH video  827k , mp4_dash container,
avc3.4D401E, 25fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_dash_https-v
ideo=827000        mp4        704x396    DASH video  827k , mp4_dash container,
avc3.4D401E, 25fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_dash-video=1570
000                mp4        704x396    DASH video 1570k , mp4_dash container,
avc3.64001F, 50fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_dash_https-vide
o=1570000          mp4        704x396    DASH video 1570k , mp4_dash container,
avc3.64001F, 50fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_dash-video=1
570000             mp4        704x396    DASH video 1570k , mp4_dash container,
avc3.64001F, 50fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_dash_https-v
ideo=1570000       mp4        704x396    DASH video 1570k , mp4_dash container,
avc3.64001F, 50fps, video only
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_hls-1013
                   mp4        704x396    1013k , avc1.4D401E@ 827k, 25.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_hls_https-1013
                   mp4        704x396    1013k , avc1.4D401E@ 827k, 25.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_hls-1013
                   mp4        704x396    1013k , avc1.4D401E@ 827k, 25.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_hls_https-10
13                 mp4        704x396    1013k , avc1.4D401E@ 827k, 25.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_hls-1800
                   mp4        704x396    1800k , avc1.64001F@1570k, 50.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_akamai_world_hls_https-1800
                   mp4        704x396    1800k , avc1.64001F@1570k, 50.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_hls-1800
                   mp4        704x396    1800k , avc1.64001F@1570k, 50.0fps, mp4
a.40.2@128k
stream-nonuk-iptv_streaming_concrete_combined_sd_mf_limelight_world_hls_https-18
00                 mp4        704x396    1800k , avc1.64001F@1570k, 50.0fps, mp4
a.40.2@128k (best)
```
😃 (FWIW, the returned formats are from an _overseas_ IP address...)

**Comment by ajj8** [comment]
Fixed in #23415

**Comment by ichux** [comment]
@azev 

I posted a comment here https://github.com/ytdl-org/youtube-dl/issues/25023#issuecomment-619694294

Do reference that and it will work for you. However, the subtle difference is that you should look for a URL with *master.mpd* in its string, e.g.

`youtube-dl https://vod-dash-ww-live.akamaized.net/usp/auth/vod/piff_abr_full_sd/bd909e-p07c6sb9/vf_p07c6sb9_82310bda-114a-4e05-b503-8dc0985cde66.ism/pc_hd_abr_v2_dash_master.mpd?__gda__=1587979639_363c73bc5d82e3bb5c188cb61e34d116
`

I tried that and it worked for me. I was able to download the content
